### PR TITLE
Prefect agent

### PR DESCRIPTION
**Pull request by gabcoyne**
First cut for prefect1.0 agent helm chart

### CODE REVIEW

**Review by zzstoatzz — APPROVED** [review]

File: k8s/helm/prefect-agent/templates/secrets.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: prefect-config
+type: Opaque
```
Comment (jamiedick) [inline k8s/helm/prefect-agent/templates/secrets.yaml:5]: may want to make this optionally created.  some people may want to use an existing secret within their cluster

File: k8s/helm/prefect-agent/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          {{- if .Values.jobTemplateFilePath }}
+          - "prefect agent kubernetes start --job-template {{ .Values.jobTemplateFilePath }}"
+          {{- else }}
+          - "prefect agent kubernetes start"
+          {{- end }}
+        env:
+          - name: PREFECT__CLOUD__AGENT__AUTH_TOKEN
+            valueFrom:
+              secretKeyRef:
+                name: prefect-config
```
Comment (jamiedick) [inline k8s/helm/prefect-agent/templates/deployment.yaml:61]: allow for this to be the default, but be overridden if an existing secret name is provided

File: k8s/helm/prefect-agent/values.yaml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+imagePullSecrets: []
+# image configures the container image for the agent deployment
+image:
+  name: prefecthq/prefect
+  tag: latest
+  pullPolicy: Always
+  pullSecrets: []
+  # - name: "secret_name"
+
+
```
Comment (jamiedick) [inline k8s/helm/prefect-agent/values.yaml:30]: secret name!

**Review by jamiedick — COMMENTED** [review]
all looks good to me -- would suggest adding the ability to create the secret (by default) but override with an existing secret if the secret name (and key?) is provided.

**Comment by gabcoyne** [comment]
Closing as moved to prefect-helm repo

**Pull request closed without merging**